### PR TITLE
Add contribution graph to release report.

### LINKED ISSUE #1469: Add Contributions git graph to cover page
programmatically add the git contributions graph to the cover sheet. This should include the amount of time between releases.

### PR DESCRIPTION
**Pull request by brianjp93**
- fixes #1469

### CODE REVIEW

**Comment by brianjp93** [comment]
image

**Review by rbbeeston — COMMENTED** [review]
LGTM, can be merged when @GregKaleka signs off on it

File: libraries/forms.py [hunk 1]
```diff
@@ -501,6 +505,111 @@ def _get_library_versions(self, library_order, version):
...
+            .values("date")
+            .annotate(count=Count("id"))
+        )
+        counts_by_date = {x["date"]: x["count"] for x in count_query}
+
+        graph = Graph()
+        # The start date is the release date of the previous version
+        # The end date is one day before the release date of the current version
+        start: date = prior_version.release_date
+        end: date = version.release_date - timedelta(days=1)
```
Comment (GregKaleka) [inline libraries/forms.py:597]: It seems like we should filter the commits by date as well so we're not handling _all_ commits - just the ones we want to graph. Does this make sense? I haven't tested this.

Also, I don't think we need to use `TruncDate`. It's minimal, but I think there's a slight performance benefit to using the `__date` transform instead.

```suggestion
        # The start date is the release date of the previous version
        # The end date is one day before the release date of the current version
        start: date = prior_version.release_date
        end: date = version.release_date - timedelta(days=1)
        count_query = (
            Commit.objects.filter(
                library_version__version=version,
                committed_at__range=(start, end)),
            )
            .values("committed_at__date")
            .annotate(count=Count("id"))
        )
        counts_by_date = {x["committed_at__date"]: x["count"] for x in count_query}

        graph = Graph()
```

**Review by GregKaleka — COMMENTED** [review]
Very cool! A couple potential (small) optimization suggestions.

File: libraries/forms.py (hunk 1 above)
Comment (brianjp93) [reply libraries/forms.py:597]: > It seems like we should filter the commits by date as well so we're not handling all commits - just the ones we want to graph. Does this make sense?

Yes, that makes sense.  I think that querying for `library_version__version=version` is the way to do that filtering.  I think it's possible for commits which were created outside of the release date range to be merged into a future release, so the date filter may not be correct.  Does that sound right?

I'll use the __date transform.

Comment (GregKaleka) [reply libraries/forms.py:597]: Oh yeah that makes sense. They should line up. 👍

**Review by GregKaleka — APPROVED** [review]
LGTM

**Comment by GregKaleka** [comment]
Looks like you need to run `pre-commit run -a`. Then please run `pre-commit install` so this happens automatically on commit!

**Pull request merged**